src/transformers/models/encodec/modeling_encodec.py

```diff
@@ -801,11 +822,12 @@ def forward(
         if audio_scales is not None and audio_codes is None:
             raise ValueError("You specified `audio_scales` but did not specify the `audio_codes`")
 
+        commitment_loss = None
         if audio_scales is None and audio_codes is None:
-            audio_codes, audio_scales = self.encode(input_values, padding_mask, bandwidth, False)
+            audio_codes, audio_scales, commitment_loss = self.encode(input_values, padding_mask, bandwidth, False)
```

Comment:
I believe we can just make the compute_loss type Optional in this case, which should allow for backwards compatibility.

Looking at some of the other models, many of them use a generic Tuple[Tensor] type for when return_dict = True. I think we can make the new third Optional parameter take on this type too, to future-proof the return type.

What do you think?

Reply:
Technically, it would still break any existing unpacking unless they caller specifically adds in the indices to unpack (ie. audio_codes, audio_scales = output_tuple[:2]). 

With that said, I went through some of the other models, and that seems to be the convention.

Out of curiosity, is the purpose of return_dict just for backwards compatibility? Would it be possible to just deprecate the return_dict? I believe JIT already supports dataclasses return types.

Reply:
It's this breaking of the unpacking that I'm a bit worried about - this would break the model for existing users. Most people won't be looking at the source code, so won't be aware of the change.

Maybe it's easier to add the loss computation at the top level model? Then we can add a `loss` entry to the `EncodecOutput` and won't break the encode method?

Reply:
I see. I will give that a try, and see how that works out.

Reply:
Yep adding at the top level is probably better! I don't have other ideas on how to make it compatible 😞